FT0 update (raw, digit writer/reader, trigger input)

afurs opened the pull request:
1. Some changes in raw event structures(from FEE group)
2. Changes and fixes in digit classes
3. Trigger input added to FT0 workflow
4. Redundant digit writer removed
5. Digit reader update(trigger input added)

Commit:
New features/changes in RawEventData, and some fixes in digits

Commit:
redundant digit writer removed, trigger inputs added to writer/reader and raw2digit

Commit:
some minor fixes

AllaMaevskaya commented:
Please merge it with recent dev

Commit:
Some fixes in RawEventData

mslupeck commented:
Hi @afurs ,

I think the checkcode is complaining about coding style (missing readability brackets). It fails with the following:

```
/O2/5728/0/Detectors/FIT/FT0/workflow/src/DigitReaderSpec.cxx:56:20: error: statement should be inside braces [readability-braces-around-statements]
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/O2/5728/0/Detectors/FIT/FT0/workflow/include/FT0Workflow/RawReaderFT0.h:51:32: error: statement should be inside braces [readability-braces-around-statements]
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/O2/5728/0/Detectors/FIT/FT0/workflow/include/FT0Workflow/RawReaderFT0.h:65:32: error: statement should be inside braces [readability-braces-around-statements]
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/O2/5728/0/Detectors/FIT/FT0/workflow/include/FT0Workflow/RawReaderFT0.h:75:32: error: statement should be inside braces [readability-braces-around-statements]
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/O2/5728/0/Detectors/FIT/FT0/workflow/include/FT0Workflow/RawReaderFT0.h:82:32: error: statement should be inside braces [readability-braces-around-statements]
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/O2/5728/0/Detectors/FIT/FT0/workflow/include/FT0Workflow/RawReaderFT0.h:51:32: error: statement should be inside braces [readability-braces-around-statements]
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/O2/5728/0/Detectors/FIT/FT0/workflow/include/FT0Workflow/RawReaderFT0.h:65:32: error: statement should be inside braces [readability-braces-around-statements]
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/O2/5728/0/Detectors/FIT/FT0/workflow/include/FT0Workflow/RawReaderFT0.h:75:32: error: statement should be inside braces [readability-braces-around-statements]
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/O2/5728/0/Detectors/FIT/FT0/workflow/include/FT0Workflow/RawReaderFT0.h:82:32: error: statement should be inside braces [readability-braces-around-statements]
--
```

Commit:
Some fixes for RawReaderFT0

afurs commented:
Hello @mslupeck,
Thanks! I fixed it.

shahor02 commented on Detectors/FIT/FT0/workflow/src/DigitReaderSpec.cxx:
```diff
@@ -49,14 +49,17 @@ void DigitReader::run(ProcessingContext& pc)
     mTree->SetBranchAddress("FT0DIGITSMCTR", &plabels);
   }
+
   mTree->GetEntry(0);
```
Just noticed that you read only event 0.  There is no particular reason to have this limitation: Note that if the decoded digits are sent to the digitWritier, it will store them as 1 entry per TF.
Could you modify it to work like https://github.com/AliceO2Group/AliceO2/blob/9f9fc26cffc8bb6be5f902ac5fe0fd689b3f6b23/Detectors/FIT/FT0/workflow/src/RecPointReaderSpec.cxx#L46-L56 ?

afurs replied:
Ok, I only added trigger input to reader, and decided to keep other parts of code untouched. But I can modify it.

Commit:
DigitReader updated, TrgInput accumulation in vector moved to DigitBlockFT0, Digit classes version increased

DigitReader updated, TrgInput accumulation in vector moved to DigitBlockFT0, Digit classes version increased

Commit:
DigitReader updated, TrgInput accumulation in vector moved to DigitBlockFT0, Digit classes version increased

AllaMaevskaya commented:
I did not find TRIGGERINPUT branch in reconstructed o2_ft0digits.root

shahor02 commented:
> I did not find TRIGGERINPUT branch in reconstructed o2_ft0digits.root

It is not supposed to be there: ``o2_ft0digits.root`` is written when the digits from decoding are stored, the triggerInputs are not encoded. They are meant to be sent during the digitization and stored in in simulated `ft0digits.root` so that CTP could access them for checks. But they don't go to rec. data.

AllaMaevskaya reviewed (approved):
OK

shahor02 reviewed (approved):
Thanks for changing also the reader! All CI failures are unrelated, squashing/merging

afurs commented:
@AllaMaevskaya ,  @shahor02 ,
Unfortunately, one line of the code https://github.com/afurs/AliceO2/blob/18d0a64beb8ec0a0af791e776e51033fff02234d/Detectors/FIT/FT0/workflow/src/FT0Workflow.cxx#L42
should be changed to  
`    specs.emplace_back(o2::ft0::getFT0DigitWriterSpec(false, !disableTrgInput));`
Should I prepare new commit and PR for this? Or is it possible to fix it in this PR?

shahor02 commented:
Hi @afurs 
This was already merged, just open a PR with fix.

shahor02 commented:
I will do this
